pkg/inject/sidecar_test.go

```diff
@@ -194,6 +194,63 @@ func TestInjectSidecarWithEnvVarsOverridePropagation(t *testing.T) {
 	assert.Contains(t, dep.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{Name: envVarServiceName, Value: "testapp.default"})
 }
 
+func TestInjectSidecarWithEnvFromK8sAppName(t *testing.T) {
+	// prepare
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "my-instance"})
+	envFromconfigMaps := []corev1.ConfigMap{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-config",
+			Namespace: "default",
+		},
+		Data: map[string]string{
+			envVarServiceName: "jaeger,b3,w3c",
+			envVarPropagation: "test-service",
+		},
+	}}
+	dep := depEnvFromConfigRef(map[string]string{}, map[string]string{
+		"app":                    "noapp",
+		"app.kubernetes.io/name": "testapp",
+	}, []corev1.EnvFromSource{{ConfigMapRef: &corev1.ConfigMapEnvSource{LocalObjectReference: corev1.LocalObjectReference{Name: "test-config"}}}})
+
+	// test
+	dep = Sidecar(jaeger, dep, WithEnvFromConfigMaps(envFromconfigMaps))
+
+	// verify
+	assert.Len(t, dep.Spec.Template.Spec.Containers, 2)
+	assert.Len(t, dep.Spec.Template.Spec.Containers[0].Env, 0)
+	assert.Len(t, dep.Spec.Template.Spec.Containers[0].EnvFrom, 1)
+	actualConfigName := dep.Spec.Template.Spec.Containers[0].EnvFrom[0].ConfigMapRef.Name
+	assert.Contains(t, "test-config", actualConfigName)
+}
+
+func TestInjectSidecarWithoutEnvFromK8sAppName(t *testing.T) {
```

Comment:
Generally, these are sub-tests or table driven tests written under one parent test since, these are not testing separate functionalities. Here, we are testing different scenarios for the same functionality. You can cross-check in the codebase if they are using table driven tests or subtests for such unit tests.

Reply:
Yes, @ApsTomar I checked in the code, and there is not any parent test case which is covering these scenarios. In the same sidecar_test.go they have returned other test cases for the same function in a similar sub-tests manner.